File: src/backend/NHP_Domain/Entities/Reservation.cs
namespace NHP_Domain.Entities;

public class Reservation
{
    public long Id { get; set; }
    public int NumeroReservation { get; set; }
    public DateTime ArriveeReservation { get; set; }
    public DateTime DepartReservation { get; set; }
    public int ParticipantReservation { get; set; }
    public long PrixReservation { get; set; }
    public String EtatReservation { get; set; }
    public int CartePaiement { get; set; }
    public String StatutPaiement { get; set; }
    
    //ManyToOne
    public virtual Remise? Remise { get; set; }
    public virtual Logement? Logement { get; set; }
    public virtual Utilisateur? Utilisateur { get; set; }

    public override string ToString()
    {
        return $"numeroReservation : {NumeroReservation} +" +
               $"arriveeReservation : {ArriveeReservation} " +
               $"departReservation : {DepartReservation} " +
               $"participantReservation : {ParticipantReservation} " +
               $"prixReservation : {PrixReservation} " +
               $"etatReservation : {EtatReservation} " +
               $"cartePaiement : {CartePaiement} " +
               $"statutPaiement : {StatutPaiement} " +
               $"remise : {Remise} " +
               $"logement : {Logement} " +
               $"utilisateur : {Utilisateur} ";
    }
}

File: src/frontend/test/src/domain/entities/Reservation.ts
import type { IRemise } from '@/domain/entities/Remise';
import type { ILogement } from '@/domain/entities/Logement';
import type { IUtilisateur } from '@/domain/entities/Utilisateur';

export interface IReservation {
    id: number;
    numeroReservation: number;
    arriveeReservation: string; 
    departReservation: string;
    participantReservation: number;
    prixReservation: number;
    etatReservation: string;
    cartePaiement: number;
    statutPaiement: string;
    remise?: IRemise;
    logement?: ILogement;
    utilisateur?: IUtilisateur;
}

export class Reservation implements IReservation {
    constructor(
        public id: number,
        public numeroReservation: number,
        public arriveeReservation: string,
        public departReservation: string,
        public participantReservation: number,
        public prixReservation: number,
        public etatReservation: string,
        public cartePaiement: number,
        public statutPaiement: string,
        public remise?: IRemise,
        public logement?: ILogement,
        public utilisateur?: IUtilisateur
    ) {}
}